### PR TITLE
Dev resources

### PR DESCRIPTION
**Pull request by vEpiphyte**
Add a contribution.rst file; including style guide notes, doc generation and contribution notes.  Also add a nose test runner.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -121,8 +121,8 @@ In order to contribute to the project, do the following:
         ./testrunner.sh
         ./testrunner.sh synapse.tests.your_test_file
 
-#. Rebase your feature branch ontop of the latest master branch of the Vertex
-   Project Synapse repository. This may require you add the Vertex Project
+#. Rebase your feature branch on top of the latest master branch of the Vertex
+   Project Synapse repository. This may require y ou to add the Vertex Project
```
Comment (vertexmc) [inline CONTRIBUTING.rst]: `y ou to` --> `you to`

File: docs/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+* The project is not currently strictly PEP8 compliant.  Compliant sections
+  include the following:
+
+  - `Whitespace in Expressions and Statements <https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements>`_.
+
+* Please keep line lengths under 120 characters.
+* Use single quotes for docstrings, not double quotes.
+* Use Napoleon format when writing docstrings. This format is very readable
+  and will allow type hinting for IDE users. Use the Google format docstring
```
Comment (vertexmc) [inline docs/contributing.rst]: I think we should just say to use the Google style here as the Google style uses Napoleon. I think it is confusing to say to use one style and to use another style.

File: docs/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  format.
+
+  - More information about Napoleon can be found `here <https://sphinxcontrib-napoleon.readthedocs.io>`_.
+  - Google style typically has the summary line after the opening ''' marker.
+    Place this summary value on the new line following the opening ''' marker.
+  - Synapse as a project is not written using the Napoleon format currently
+    but all new modules should adhere to that format.
+
+* Imports should be in order of shortest to longest import, not alphabetical
+  order. Imports should be ordered starting from the Python standard library
```
Comment (whippit) [inline docs/contributing.rst]: secondary sort is alpha when lengths match.

File: docs/contributing.rst (hunk 2 above)
Comment (invisig0th) [inline docs/contributing.rst]: Is there any other pydoc instances where google has a style ( ie, do they also have a pydoc style for when a project is *not* using sphinx/napoleon?  It feels to me like it's correct to say the "google dialect" of sphinx/napoleon or some such, but this is the first i've looked into it :D

File: docs/contributing.rst [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  - foo_the_bar() is not acceptable.
+
+* Private methods should be marked as such with a proceeding underscore.
+
+  - privateInternalThingDontUseMe() is not acceptable.
+  - _internalThing() is acceptable.
+
+* Logging should be setup on a per-module basis, with loggers created using
+  calls to logging.getLogger(__name__).  This allows for module level control
+  of loggers as neccesary.
```
Comment (invisig0th) [inline docs/contributing.rst]: could we insert "code block" examples for this ( and maybe a few others? )

File: docs/contributing.rst (hunk 2 above)
Comment (vEpiphyte) [inline docs/contributing.rst]: Added link to their style reference.

File: docs/contributing.rst (hunk 3 above)
Comment (vEpiphyte) [inline docs/contributing.rst]: fixed

File: docs/contributing.rst (hunk 4 above)
Comment (vEpiphyte) [inline docs/contributing.rst]: Done

File: docs/contributing.rst [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+      # Get the module level logger
+      logger = logging.getLogger(__name__)
+      # Do this - it only forms the final string if the message is
+      # actually going to be logged
+      logger.info('I am a message from %s about %s', 'bob', 'a duck')
+      # NOT this - it performs the string format() call regardless of
+      # whether or not the message is going to be logged.
+      logger.info('I am a message from {} about {}'.format('bob', 'a duck'))
+
+* It may be neccesary from time to time to include non-ASCII characters. Use
```
Comment (vertexmc) [inline docs/contributing.rst]: I think I have also had to import this:
```from __future__ import absolute_import,unicode_literals```

**Pull request merged**